File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
        int n = arr.size();
        int l  = 0 , h = n-1 ;
        while(l < h){
            if(arr[l]+arr[h] == target){
                return {l+1, h+1};
            }
            else if(arr[l] + arr[h] > target)
                h--;
            else
                l++;
        }   
        return {};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        int []ans = new int[2];
        int l = 0 , h = arr.length-1;
        while(l < h){
            int sum = arr[l] + arr[h];
            if(sum == target){
                ans[0] = l+1;
                ans[1] = h+1;
                break;
            }
            else if(sum > target){
                h--;
            }
            else{
                l++;
            }
        }
        return ans;
    }
}